pkg/quarterdeck/auth.go

```diff
@@ -81,6 +95,19 @@ func (s *Server) Register(c *gin.Context) {
 		return
 	}
 
+	// Link the user's organization to the project by creating a database record.
+	// This is a critical security step to ensure that users in other organizations
+	// cannot issue requests with this project ID.
+	op := &models.OrganizationProject{
+		OrgID:     org.ID,
+		ProjectID: projectID,
+	}
+	if err = op.Save(ctx); err != nil {
+		log.Error().Err(err).Msg("could not link organization to project")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not process registration"))
+		return
+	}
```

Comment:
Note that this executes in a separate transaction from `user.Create`. I'd like to make the `ProjectID` optional - so having a separate transaction is perfectly fine; but it can create the possibility that a user/organization is created but an `OrganizationProject` is not created.